src/main/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactory.java

```diff
@@ -384,4 +429,111 @@ private static int qid(int streamIdx, int spatialIdx, int temporalIdx,
         return streamIdx * spatialLen * temporalLen
             + spatialIdx * temporalLen + temporalIdx;
     }
+
+    /**
+     * Describes a single secondary ssrc and its group semantics to
+     * its primary ssrc (e.g. FID or FEC-FR)
+     */
+    private static class SecondarySsrc
+    {
+        public long ssrc;
+        public String type;
+
+        public SecondarySsrc(long ssrc, String type)
+        {
+            this.ssrc = ssrc;
+            this.type = type;
+        }
+    }
+
+    /**
+     * Groups a set of secondary ssrcs for a single primary ssrc (e.g. its
+     * rtx and fec ssrcs)
+     */
+    private static class SecondarySsrcs
+        implements Iterable<SecondarySsrc>
+    {
+        public List<SecondarySsrc> secondarySsrcs;
+
+        public SecondarySsrcs(List<SecondarySsrc> secondarySsrcs)
+        {
+            this.secondarySsrcs = secondarySsrcs;
+        }
+
+        @Override
+        public Iterator<SecondarySsrc> iterator()
+        {
+            return secondarySsrcs.iterator();
+        }
+    }
+
+    /**
+     * Describes one or more primary video ssrcs for a single
+     * {@link MediaStreamTrackDesc}
+     */
+    private static class TrackSsrcs
```

Comment:
What is the purpose of this class? Couldn't we just use a `List`?

Reply:
this makes things a lot more readable, since we use it in a list. so this avoids having List<List<Long>> and instead gives List<TrackSsrcs> which is a lot clearer.

Reply:
I see, how about extending a `List` then?

Reply:
i gave this a shot but ran into a couple problems:

1) it has to derive from a list implementation, not just `List` (or else i'd have to implement the abstract `List` methods, which doesn't really seem worth it), which doesn't feel right to me: nothing about this requires it being an `ArrayList`, for example.
2) It specifically would derive from `ArrayList<Long>` (since we want it to only hold ssrcs), which breaks some of the ways I'm currently constructing it.  `Arrays.asList(group.getSources().get(0).getSSRC())`, for example, evaluates to List<T> and doesn't allow assigning to a `TrackSsrcs` object which derives from `ArrayList<Long>`